src/Features/Core/Portable/SyncNamespaces/ISyncNamespacesService.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.SyncNamespaces
+{
+    internal interface ISyncNamespacesService : ILanguageService
+    {
+        Task<Solution> SyncNamespacesAsync(ImmutableArray<Project> projects, CancellationToken cancellationToken);
```

Comment:
Since it was of some confusion in my PR, worth calling out that this syncs the namespace to the folder structure.